src/main/java/software/amazon/payloadoffloading/S3Dao.java

```diff
@@ -61,13 +61,17 @@ public String getTextFromS3(String s3BucketName, String s3Key) {
     }
 
     public void storeTextInS3(String s3BucketName, String s3Key, SSEAwsKeyManagementParams sseAwsKeyManagementParams,
-                              String payloadContentStr, Long payloadContentSize) {
+                              CannedAccessControlList cannedAccessControlList, String payloadContentStr, Long payloadContentSize) {
```

Comment:
S3Dao methods are getting complex as we introduce new properties. I think we should consider moving SSEAwsKeyManagementParams and CannedAccessControlList as class properties. Also, payloadContentSize can be abstracted from the user.

Reply:
Moved SSEAwsKeyManagementParams and CannedAccessControlList as properties in S3Dao